### ISSUE TITLE
Skip only faulty entries

### ISSUE THREAD

**Issue by moewew** [plk/biber#270]
Inspired by https://tex.stackexchange.com/q/497829/35864 I am wondering if Biber could be slightly more tolerant about malformed `.bib` entries.

Currently a malformed `.bib` entry like in the following example completely aborts the Biber run and leaves an empty `.bbl` file.

```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber]{biblatex}


\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@book{appleby,
  author  = {Humphrey Appleby},
  title   = {On the Importance of the Civil Service},
  date    = {1980},
}
@book{bppleby,
  author  = {Humphrey Bppleby}
  title   = {On the Importance of the Civil Service},
  date    = {1981},
}
@book{cppleby,
  author  = {Humphrey Cppleby},
  title   = {On the Importance of the Civil Service},
  date    = {1982},
}
\end{filecontents}

\addbibresource{\jobname.bib}


\begin{document}
\cite{appleby,bppleby,cppleby}
\printbibliography
\end{document}
```

The error message by Biber
```
ERROR - BibTeX subsystem: <file>, line 12,
        syntax error: found "title", expected end of entry ("}" or ")") (skipping to next "@")
```
seems to suggest that only the faulty entry is skipped. I suspect the message comes from `btparse`, so I assume that `btparse` would have the ability to just skip the faulty entry and press on, that's why I'm asking if this would also be possible for Biber.

Also related https://github.com/plk/biber/issues/261

**Comment by plk** [comment]
Fix in DEV. Note that this error means that it skips from where it finds the error to the next entry, leaving an entry with incomplete data. It doesn't skip the entry and there is no way to do that. `btparse` doesn't do that and due to the old-fashioned way `btparse` handles STDERR, it can't be caught in real-time so that `biber` could skip the whole entry.

**Comment by moewew** [comment]
Ah, it would have been a nice added bonus if had been possible to completely remove the entry as that would have been more visible for the user (surprisingly few people read the `.blg` or Biber output due to their editor setup), but if it is not possible to do that we at least emulate BibTeX in that behaviour.

As in #261 I think it would be much nicer to display the
```
[341] Utils.pm:300> WARN - BibTeX subsystem: C:\Users\Moritz\AppData\Local\Temp\34LxXYaFt1\wuljdhsjsh.bib_12600.utf8, line 12, syntax error: found "title", expected end of entry ("}" or ")") (skipping to next "@")
```
in the log as a real ERROR. Is that possible? (It seems that the only change here was `biber_error`, to `biber_warn`, so I assume some work would be needed for an error that doesn't abort processing...)

**Comment by plk** [comment]
It's a pain that bibtex/btparse don't report the entries with errors, just the line numbers in the file because they are basically just streaming the file.
Can you try DEV again now for the error?

**Comment by moewew** [comment]
Thank you. The example works as I expected now.

It would be great if there was some way to identify the problematic entry, but BibTeX is not better in that regard.

Previously I found it more useful to plough on despite the error, but I can definitely see the appeal in completely aborting the Biber run and producing an empty `.bbl` to tell the user that something is off (a half-processed entry is much less visible). What do you think?

**Comment by plk** [comment]
Well, there is already the `biber` option `--nodieonerror` so people could choose. Currently, the fix overrides this option but I can easily make it obey that option - the default is always to die on errors.

**Comment by moewew** [comment]
Ah yes, given all that I guess I'll change my mind and vote for making Biber die on these kind of errors by default again and have the user override that explicitly with `--nodieonerror`.

**Comment by plk** [comment]
Fixed in latest 2.13 release.
